cmd/scollector/collectors/stream.go

```diff
@@ -0,0 +1,82 @@
+package collectors
+
+import (
+	"reflect"
+	"runtime"
+	"sync"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type StreamCollector struct {
```

Comment:
I felt if any other future collectors would follow the same pattern having a StreamCollector in addition of the IntervalCollector would be beneficial. I'll gladly remove it and attach the collector in main.go if you wish though.